xknx/management/procedures.py

```diff
@@ -184,3 +184,56 @@ async def nm_invididual_address_write(
             tpci=tpci.TDataConnected(sequence_number=seq_num),
         )
         await xknx.cemi_handler.send_telegram(telegram)
+
+
+async def nm_individual_address_serial_number_read(
+    xknx: XKNX, serial: bytes, timeout: float = 3
+) -> IndividualAddress | None:
+    """Read individual address from device with specified serial number."""
+
+    # initialize queue or event handler gathering broadcasts
+    async with xknx.management.broadcast() as bc_context:
+        broadcast_telegram = Telegram(
+            destination_address=GroupAddress("0/0/0"),
+            source_address=xknx.current_address,
+            payload=apci.IndividualAddressSerialRead(serial=serial),
+        )
+        await xknx.management.send_broadcast(broadcast_telegram)
+        async for result in bc_context.receive(timeout=timeout):
+            if isinstance(result.payload, apci.IndividualAddressSerialResponse):
+                return result.source_address
+
+    return None
+
+
+async def nm_individual_address_serial_number_write(
+    xknx: XKNX, serial: bytes, individual_address: IndividualAddressableType
+) -> None:
+    """Write individual address to device with specified serial number."""
+    individual_address = IndividualAddress(individual_address)
+    await xknx.management.send_broadcast(
+        Telegram(
+            destination_address=GroupAddress("0/0/0"),
+            payload=apci.IndividualAddressSerialWrite(
+                address=individual_address, serial=serial
+            ),
+        )
+    )
```

Comment:
The specs say 
> The procedure shall ensure that the assigned Individual Address is unique.

, but don't have any means of doing so in the procedure. Is it expected that the management client knows and uses `nm_individual_address_check` by themselves before?

Reply:
What I implemented is based on the sequence diagram contained in the spec: 
![image](https://github.com/XKNX/xknx/assets/8012364/1fff63bc-63d9-48df-a862-6bb5018fe93e). I think that the device issuing this procedure should indeed first check the availability with `nm_individual_address_check`.